android/src/app/organicmaps/widget/placepage/PlacePageView.java

```diff
@@ -253,14 +250,34 @@ public void onAttach(@NonNull Context context)
   }
 
   @Override
-  public void onDestroyView()
+  public void onResume()
   {
-    super.onDestroyView();
-    detachCountry();
+    super.onResume();
+    mViewModel.getMapObject().observe(requireActivity(), this);
+    LocationHelper.INSTANCE.addListener(this);
+  }
+
+  @Override
+  public void onPause()
+  {
+    super.onPause();
+    // Unsubscribe from events as soon as the fragment becomes inactive
+    // to prevent unwanted side effects
     mViewModel.getMapObject().removeObserver(this);
     LocationHelper.INSTANCE.removeListener(this);
   }
 
+  @Override
+  public void onStop()
+  {
+    super.onStop();
+    // Safely detach the country once the fragment is hidden from the user
+    // It is safer to call this here than in onPause as the app could go from onPause to
+    // onResume without killing the fragment.
+    // In this case we would not want to detach the country.
+    detachCountry();
```

Comment:
Storage callbacks are using a lot of unnecessary CPU. Why it can't be detached in onPause and attached again in onResume?

Reply:
onPause may be called while the app is still visible to the user, while onStop is only called when the activity is fully covered by another (after onPause). We should never call CPU intensive functions inside onPause because it could slow down the transition animation (see https://developer.android.com/reference/android/app/Activity.html#onPause()).

So here the country is only attached when the current map object changes, and detached when the activity/fragment stops.